driver/src/main/java/org/neo4j/driver/internal/NetworkSession.java

```diff
@@ -174,10 +175,14 @@ public synchronized Transaction beginTransaction()
         return beginTransaction( mode );
     }
 
+    @Deprecated
     @Override
     public synchronized Transaction beginTransaction( String bookmark )
     {
-        lastBookmark = bookmark;
+        if ( bookmark != null )
```

Comment:
`setBookmark` will still have to support `null` as we don't have a separate `clearBookmark` method. But using the setter inside the `if` block is a good idea.